Formula/kde-kplotting.rb

```diff
@@ -0,0 +1,64 @@
+class KdeKplotting < Formula
+  desc "Lightweight plotting framework"
+  homepage "https://api.kde.org/frameworks/kplotting/html/index.html"
```

Comment:
Why is this called `kde-kplotting` rather than just `kplotting`? Are there any libraries inside or outside Homebrew called `kplotting`?

Reply:
This was discussed before: https://github.com/Homebrew/homebrew-core/pull/56279#issuecomment-643843896

Reply:
@smillerdev can you explain the original motivation for these prefixes? If it wasn't you who decided this, can you CC `@Homebrew/core` to ask? Thanks!

Reply:
https://github.com/Homebrew/homebrew-core/pull/55120#issuecomment-632688209

There are a couple that aren't clearly KDE and those happened to be the first to be migrated so I voted for kde prefixes. For ki18n it was on the edge for me, but with things like this I'm not sure it's needed anymore. Although I haven't really thought about that with other merges.

Reply:
Taking one such KDE framework which name isn't so obvious, it looks like vast majority of repositories let the name be unchanged after all: https://repology.org/project/attica/versions

I suppose the existing formulas can be renamed if needed, and a reversing Alias added for backwards compatibility?

Reply:
Homebrew generally only ships the latest version of software, so unless `kf5-*` formulae are really popular and are going to maintained for a while upstream there will only ever be one version of `kf-*` so it might as well be `kde-*`

Reply:
These are libraries, which many apps depend on. They will have ABI and API breakages, so many, if not majority of these apps won't update right away. There's definitely gonna be a period of time you'll be maintaining those simultaneously.

This will be very similar to how current qt@5 and qt@6 situation looks like.

Reply:
> I suppose the existing formulas can be renamed if needed

Yes, this would be ideal, thanks. Would you be able to open a PR to do that @wrobelda? See https://docs.brew.sh/Rename-A-Formula and ask here if you need any help.

> and a reversing Alias added for backwards compatibility?

Good idea but: this is unnecessary as it's already handled automatically when we rename formulae.

> Homebrew generally only ships the latest version of software, so unless `kf5-*` formulae are really popular and are going to maintained for a while upstream there will only ever be one version of `kf-*` so it might as well be `kde-*`

Yes, this. Older versions will have to meet all these requirements: https://docs.brew.sh/Versions

Additionally, we'd rather have them called `ki18n@5` rather than `kf5-ki18n` or whatever.

> These are libraries, which many apps depend on. KF6 will have ABI and API breakages, since it will be based on Qt6. So many, if not majority of these apps won't update right away. There's definitely gonna be a period of time you'll be maintaining those simultaneously.

This applies to almost every library we package. It doesn't matter: we ship the latest versions of things and remove old versions. That's just how Homebrew works.

> This will be very similar to how current qt@5 and qt@6 situation looks like.

They have _vastly_ higher usage.

---

If it helps convince you of my non-ignorance: I used to be a KDE contributor 😉

Reply:
> This applies to almost every library we package. It doesn't matter: we ship the latest versions of things and remove old versions. That's just how Homebrew works.

That's fair, but it only works if you don't also offer Formulas for the software that leverages those libraries, doesn't it? So if say I want to have kmymoney formula upstream, and it has dependencies on kf5 libs, when kf5 gets upgraded to kf6, what would you do then? Probably not remove kmymoney? Or am I missing something?

Reply:
> So say I want to have kmymoney formula upstream, which has dependencies on kf5 libs, once kf5 gets upgraded to kf6, what would you do then? Probably not remove kmymoney? Or am I missing something?

`kmymoney` would need to vendor a kf5 formula. This is the same as with any versioned formulae we currently use.

> Agreed. So correct me if I am wrong, but we can call them ki18n for now, and if we end up having to simultaneously support both kf5 and kf6, adding a `@5` suffix like with `qt@5` will sort it out.

Exactly 👍🏻